nexus/internal/filetransfer/file_transfer_manager.go

```diff
@@ -202,13 +204,29 @@ func (fm *FileTransferManager) transfer(task *Task) error {
 	case UploadTask:
 		err = fm.fileTransfer.Upload(task)
 	case DownloadTask:
+		// Skip downloading the file if it already exists
+		if _, err := os.Stat(task.Path); err == nil {
+			return nil
+		}
+		if err := fm.ensureDownloadRootDir(task.Path); err != nil {
+			return err
+		}
```

Comment:
maybe move this logic to task creation or download. 
if we already have the file why do we create the task?
also, the dir setup seems like it should go in the download function?

Reply:
That's where it was earlier. I moved it here based on another review comment. Shall i move it back to `fileTransfer.Download()` then?